lib/inspec/profile.rb

```diff
@@ -96,6 +96,7 @@ def initialize(source_reader, options = {})
       @attr_values = options[:attributes]
       @tests_collected = false
       @libraries_loaded = false
+      @check_mode = options[:check_mode] || false
```

Comment:
Little Ruby nugget - careful with `||` and default booleans. For example, if you wanted the default to be `true` and you wrote this like:

```ruby
@check_mode = options[:check_mode] || true
```

... it could never be false! If the user passed in `false` as the value in `options[:check_mode]`, it would fail the left side of the `||` and then right side would evaluate.

I tend to use `#fetch` on `Hash` for this instead, just in case:

```ruby
@check_mode = options.fetch(:check_mode, false)
```

Reply:
Ah, thanks for the tip. Never thought of that.